tests/lexer/tstrnumlits.nim

```diff
@@ -0,0 +1,91 @@
+discard """
+  output: "[[1]]neg[[1]][[1]][[12345.67890]][[12345.67890]][[1]]times[[1]][[1]]plusneg[[1]][[1]][[22.40]][[0x5a3a]]neg[[38383839292839283928392839283928392839283.928493849385935898243e-50000]]"
+"""
+# Test tkStrNumLit
+
+type
+  S = distinct string
+
+proc wrap(number: string): S =
+  result = ("[[" & number & "]]").S
+
+template twrap(number: string): untyped =
+  wrap(number)
+
+proc extraContext(): S =
+  22.40wrap
+
+# if a user is writing a numberic library, they might write proc similar to:
+
+proc `*`(left, right: S): S =
+  result = (left.string & "times" & right.string).S
+
+proc `+`(left, right: S): S =
+  result = (left.string & "plus" & right.string).S
+
+proc `-`(number: S): S =
+  result = ("neg" & number.string).S
+
+proc `$`(number: S): string =
+  result = number.string
+
+const
+  likeInteger    = 1wrap
+  negative       = -1wrap
+  quoted         = 1wrap      # 1'wrap
+  likeFloat      = 12345.67890wrap
+  quotedFloat    = 12345.67890wrap # 12345.67890'wrap
+  smushed        = 1wrap*1wrap
+  borderCase     = 1wrap + -1wrap
+  usingTemplate  = 1twrap
+  inProc         = extraContext()
+  hex            = 0x5a3awrap  # Note: starting a suffix with the letters `a` to `f` is NOT a good idea
+  reallyBig      = -38383839292839283928392839283928392839283.928493849385935898243e-50000wrap
+
+stdout.write(likeInteger.string)
+stdout.write(negative.string)
+stdout.write(quoted.string)
+stdout.write(likeFloat.string)
+stdout.write(quotedFloat.string)
+stdout.write(smushed.string)
+stdout.write(borderCase.string)
```

Comment:
avoid defining symbols used just once and instead use them inline, which makes the test easier to read, eg:
doAssert string(1wrap + -1wrap) == ...

Reply:
done.